### PR TITLE
expression: implement vectorized evaluation for `builtinSubstring2ArgsSig`

### PR DESCRIPTION
**Pull request by shihongzhi**
### What problem does this PR solve? 

implement vectorized evaluation for builtinSubstring2ArgsSig , for #12106

### What is changed and how it works?

	$go test -v -benchmem -bench=BenchmarkVectorizedBuiltinStringFunc -run=BenchmarkVectorizedBuiltinStringFunc -args "builtinSubstring2ArgsSig"
	goos: darwin
	goarch: amd64
	pkg: github.com/pingcap/tidb/expression
	BenchmarkVectorizedBuiltinStringFunc/builtinSubstring2ArgsSig-VecBuiltinFunc-4         	   16394	     73338 ns/op	       0 B/op	       0 allocs/op
	BenchmarkVectorizedBuiltinStringFunc/builtinSubstring2ArgsSig-NonVecBuiltinFunc-4      	   10000	    111463 ns/op	    7232 B/op	    1024 allocs/op
	PASS
	ok  	github.com/pingcap/tidb/expression	3.119s

### Check List 

Tests 

 - Unit test

### CODE REVIEW

File: expression/builtin_string_vec.go [hunk 1]
```diff
@@ -747,11 +747,54 @@ func (b *builtinSubstringBinary2ArgsSig) vecEvalString(input *chunk.Chunk, resul
...
+	for i := 0; i < n; i++ {
+		if buf.IsNull(i) || buf2.IsNull(i) {
+			result.AppendNull()
+			continue
+		}
+
+		str := buf.GetString(i)
+		pos := nums[i]
+
+		runes := []rune(str)
```
Comment (zz-jason) [inline expression/builtin_string_vec.go]: could you add a test case for substring on a `binary(40)` column with the value `"中文字符串"`?

Comment (shihongzhi) [reply expression/builtin_string_vec.go]: > could you add a test case for substring on a `binary(40)` column with the value `"中文字符串"`?

this is `builtinSubstring2ArgsSig` , not `builtinSubstringBinary2ArgsSig`. do you mean add `char(40)`  column with the value "中文字符串" ?

**Pull request merged**